#!/usr/bin/env python
PACKAGE = "groundgrid"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("point_count_cell_variance_threshold", int_t, 0, "if a cell has at least this much points, the variance of just that cell is used instead of the variance of 3x3 or 5x5 patch", 10, 0, 30)
gen.add("max_ring", int_t, 0, "maximum velodyne ring for ground detection consideration", 256, 0, 256)
gen.add("groundpatch_detection_minimum_threshold", double_t, 0, "if the ground patch layer at a cell is below this value, a cell without the minimum point count can be classified as ground", 0.01, 0, 1.0)
gen.add("distance_factor", double_t, 0, "compensates for the geometric dilution of the point density with the distance", 0.0001, 0, 0.01)
gen.add("minimum_distance_factor", double_t, 0, "minimum value for the distance factor", 0.0005, 0, 0.01)
gen.add("miminum_point_height_threshold", double_t, 0, "Points lower than ground height + threshold are considered ground points", 0.3, 0, 1)
gen.add("minimum_point_height_obstacle_threshold", double_t, 0, "Points lower than ground height + threshold are considered ground points [m]", 0.1, 0, .2)
gen.add("outlier_tolerance", double_t, 0, "Outlier detection tolerance [m]", 0.1, -.5, .5)
gen.add("ground_patch_detection_minimum_point_count_threshold", double_t, 0, "Minimum point count for ground patch detection in percent of expected point count", 0.25, 0.01, 1.0)
gen.add("patch_size_change_distance", double_t, 0, "Distance from the center from which on the patch size is increased [m]", 20, 0, 50)
gen.add("occupied_cells_decrease_factor", double_t, 0, "Occupied cells decrease factor [100/x %]", 5.0, 1.0, 100.0)
gen.add("occupied_cells_point_count_factor", double_t, 0, "Occupied cells point count factor [100/x %]", 20, 1, 50)
gen.add("min_outlier_detection_ground_confidence", double_t, 0, "Minimum ground confidence to consider lower points an outlier (5x5 patch)", 1.25, 0, 5)
gen.add("thread_count", int_t, 0, "Maximum number of threads", 8, 1, 64)

exit(gen.generate(PACKAGE, "groundgrid", "GroundGrid"))
